collect_app/src/main/java/org/odk/collect/android/tasks/ImageUploadingTask.java

```diff
@@ -0,0 +1,87 @@
+package org.odk.collect.android.tasks;
+
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.provider.MediaStore;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.Fragment;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.exception.GDriveConnectionException;
+import org.odk.collect.android.fragments.dialogs.ProgressDialogFragment;
+import org.odk.collect.android.utilities.FileUtils;
+import org.odk.collect.android.utilities.ImageConverter;
+import org.odk.collect.android.utilities.MediaUtils;
+import org.odk.collect.android.utilities.ToastUtils;
+
+import java.io.File;
+
+import timber.log.Timber;
+
+public class ImageUploadingTask extends AsyncTask<Uri, Void, File> {
+
+    private FormEntryActivity formEntryActivity;
+
+    public ImageUploadingTask(FormEntryActivity formEntryActivity) {
+        onAttach(formEntryActivity);
+    }
+
+    public void onAttach(FormEntryActivity formEntryActivity) {
+        this.formEntryActivity = (FormEntryActivity) formEntryActivity;
+    }
+
+    public void onDetach() {
+        this.formEntryActivity = null;
+    }
+
+    @Override
+    protected File doInBackground(Uri... uris) {
+        File instanceFile = formEntryActivity.getFormController().getInstanceFile();
```

Comment:
please add a null check here (formController object may be null) we had lot's off related crashes.